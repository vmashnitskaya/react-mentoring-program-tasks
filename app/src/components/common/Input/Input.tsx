import React, {ChangeEvent, FunctionComponent} from 'react';

interface InputProps {
  type?: string;
  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
  id: string;
  value: string | number;
  label?: string;
  disabled?: boolean;
  children?: JSX.Element;
}

const Input: FunctionComponent<InputProps> = ({type = "text", onChange, children, id, value, label, disabled }) => {

  return (
      <label htmlFor={id}>
        {label}
        <input
            id={id}
            value={value}
            onChange={onChange}
            type={type}
            disabled={disabled}
        />
        {children}
      </label>
  );
};

export default Input;
