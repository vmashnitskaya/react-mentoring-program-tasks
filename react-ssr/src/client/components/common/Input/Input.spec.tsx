import React from 'react';
import { render } from '@testing-library/react';
import Input from "./Input";

describe('Input', () => {
    test('renders input snapshot', () => {
        const onChangeFunction = jest.fn();
        const {asFragment} = render(<Input id={"id"} value={'value'} onChange={onChangeFunction}/>);

        expect(asFragment()).toMatchSnapshot();
    });
});
