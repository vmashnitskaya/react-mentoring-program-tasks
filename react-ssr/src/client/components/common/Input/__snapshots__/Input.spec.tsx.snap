// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders input snapshot 1`] = `
<DocumentFragment>
  <label
    for="id"
  >
    <input
      id="id"
      type="text"
      value="value"
    />
  </label>
</DocumentFragment>
`;
